### ISSUE TITLE
Could we use this to improve MMLU capability?

### ISSUE THREAD

**Issue by YixinSong-e** [Kipok/NeMo-Skills#3]
Exciting work! Could we use this to improve MMLU and other capability?

**Comment by Kipok** [comment]
Yes, certainly! We are working on improving coding skills already.

In general, you can directly apply our pipeline to any task for which you have some training data of the form "question" / "answer" where "answer" can be automatically verified for correctness. As long as you have this, you can just write a couple of few-shot examples of the solutions you want to teach LLM to produce, modify the code that checks answer for correctness (it's currently specific for math) and then run synthetic data generation and SFT. To take the most out of our pipeline, it's best if the solutions also leverage Python code in some way.

Let us know if you have any questions about the details of this - we'd be happy to see this pipeline applied to other tasks!
